### PR TITLE
Add resource version in instance manager

### PR DESCRIPTION
**Pull request by shuo-wu**

### CODE REVIEW

File: rpc/rpc.proto [hunk 1]
```diff
@@ -34,6 +34,7 @@ message EngineSpec {
 message EngineStatus {
 	ProcessStatus process_status = 1;
 	string endpoint = 2;
+	int64 revision_counter = 3;
```
Comment (yasker) [inline rpc/rpc.proto]: Don't call it `revision_counter`, just `revision` or `resource_version`.

We're going to get rid of `revision_counter` in the replica.

Comment (shuo-wu) [reply rpc/rpc.proto]: Why do we need to get rid of it in the replica? We also need to make sure that the latest replica process status won't be overwritten by expired status.

Comment (yasker) [reply rpc/rpc.proto]: I meant in the replica data path. Totally different thing. https://github.com/longhorn/longhorn/issues/508

Comment (shuo-wu) [reply rpc/rpc.proto]: Do you mean the variable name `revision_counter` had been used in the replica?

**Pull request merged**